test/src/org/labkey/test/tests/targetedms/TargetedMSTrailingMeanAndCVTest.java

```diff
@@ -0,0 +1,177 @@
+        String toolTipText = waitForElement(qcPlotsWebPart.getBubbleContent()).getText();
+        Assert.assertEquals("Invalid tooltip", "Peptide:\n" +
+                "ATEEQLK ++, 409.7163\n" +
+                "Value:\n" +
+                "15.652\n" +
+                "Replicate:\n" +
+                "3 runs average\n" +
+                "Acquired:\n" +
+                "2013-08-09 11:39:00 - 2013-08-12 04:54:55", toolTipText);
+
+        log("Selecting multiple plot types with trailing mean");
+        qcPlotsWebPart.setQCPlotTypes(QCPlotsWebPart.QCPlotType.LeveyJennings, QCPlotsWebPart.QCPlotType.MovingRange, QCPlotsWebPart.QCPlotType.TrailingMean);
+        Assert.assertEquals("Incorrect number of plots displayed when multiple plot type are selected", 21, qcPlotsWebPart.getTotalPlotCount());
+
+        log("Verifying values with guide set");
+        createGuideSetFromTable(new GuideSet("2013/08/09 23:59", "2013-08-15 00:00:04", "Trailing mean guide set"));
+        goToProjectHome();
+        dashboard = new PanoramaDashboard(this);
+        qcPlotsWebPart = dashboard.getQcPlotsWebPart();
+        qcPlotsWebPart.setQCPlotTypes(QCPlotsWebPart.QCPlotType.TrailingMean);
+        mouseOver(qcPlotsWebPart.getPointByAcquiredDate("2013-08-14 00:44:46"));
+        toolTipText = waitForElement(qcPlotsWebPart.getBubbleContent()).getText();
+        Assert.assertEquals("Invalid tooltip of the point in guide set", "Peptide:\n" +
+                "ATEEQLK ++, 409.7163\n" +
+                "Value:\n" +
+                "15.684\n" +
+                "Replicate:\n" +
+                "3 runs average\n" +
+                "Acquired:\n" +
+                "2013-08-11 18:34:14 - 2013-08-14 00:44:46", toolTipText);
+
+        log("Verifying the count of points on the plots with guide set");
+        Assert.assertEquals("Invalid point count for all replicates", 308 , qcPlotsWebPart.getPointElements("d", SvgShapes.CIRCLE.getPathPrefix(), true).size());
+    }
+
+    @Test
+    public void testTrailingCVPlotType()
+    {
+        goToProjectHome();
+        PanoramaDashboard dashboard = new PanoramaDashboard(this);
+        QCPlotsWebPart qcPlotsWebPart = dashboard.getQcPlotsWebPart();
+        qcPlotsWebPart.setQCPlotTypes(QCPlotsWebPart.QCPlotType.TrailingCV);
+
+        log("Verifying error checking on trailing last number");
+
+        qcPlotsWebPart.setTrailingLast("-1");
+        longWait().until(ExpectedConditions.textToBePresentInElement(Locators.labkeyError.findWhenNeeded(getDriver()),
+                "TrailingCV - Please enter a positive integer (>2) that is less than or equal to total number of available runs - 47"));
+
+        qcPlotsWebPart.setTrailingLast("48");
+        longWait().until(ExpectedConditions.textToBePresentInElement(Locators.labkeyError.findWhenNeeded(getDriver()),
+                "TrailingCV - The number you entered is larger than the number of available runs. Only 47 runs are used for calculation"));
```

Comment:
We should assert that these errors don't prevent other plots from showing up. I think that was an issue during manual testing.